translate.py

```diff
@@ -102,9 +102,28 @@ def report_score(name, score_total, words_total):
             name, score_total / words_total,
             name, math.exp(-score_total/words_total)))
 
+    def report_bleu():
+        import subprocess
+        print()
+        res = subprocess.check_output(
+            "perl tools/multi-bleu.perl %s < %s" % (opt.tgt, opt.output),
+            shell=True).decode("utf-8")
+        print(">> " + res.strip())
+
+    def report_rouge():
+        import subprocess
+        res = subprocess.check_output(
+            "python tools/test_rouge.py -r %s -c %s" % (opt.tgt, opt.output),
+            shell=True).decode("utf-8")
+        print(res.strip())
```

Comment:
Could you move these `report_*()` functions out of `main()`?  And prefix them with `_` to mark them as intra-file use only.

Reply:
No Problem. But what about ```report_score()``` ? Should I move it out in the same way?

Reply:
I think you could do so for all of them. Previously there is only one `report_*()`, but now we have 3, and making them stay in `main()` would clutter the main code flow and distract readers.

Reply:
Agree with you.